nucypher/blockchain/eth/interfaces.py

```diff
@@ -231,7 +230,7 @@ def __init__(self,
         self.client = NO_BLOCKCHAIN_CONNECTION         # type: EthereumClient
         self.transacting_power = READ_ONLY_INTERFACE
         self.is_light = light
-        self.gas_strategy = gas_strategy
+        self.gas_strategy = gas_strategy or self.DEFAULT_GAS_STRATEGY
```

Comment:
So we're expecting the user of this class to ignore the type attribute and pass `None` anyway?

Reply:
No.  The type attribute annotation is inaccurate.

Reply:
I'm unsure what's the issue here. There are many situations where we don't explicitly initialize the `BlockchainInterface` with a gas strategy (e.g., a staker has a configuration file with `null` as gas strategy because none was set with `nucypher ursula init`, see #2421), and the default should be used. In current `main`, the default is set as an argument default, but we sometimes need to change it in subclasses of `BlockchainInterface`, and we have already discussed in the past how this pattern doesn't work and we've been using incorrectly (see description of 4a263940ba037d89302213b0a8679b0fdaff1b65).

If the issue is the lack of an `Optional` type hint, I've just added.